File: Programmers_level1/08. 소수 만들기.py
import math

def isPrime(num):
    for i in range(2, int(math.pow(num, 0.5))+1):
        if num % i == 0:
            return False
    else:

        return True

def solution(nums):
    answer = 0
    sum = 0
    
    for i in range(len(nums)):
        for j in range(i+1, len(nums)):
            for k in range(j+1, len(nums)):
                sum = nums[i] + nums[j] + nums[k]
                if isPrime(sum):
                    answer += 1

    return answer

File: Programmers_level1/08. 소수 만들기.swift
import Foundation
func isPrime(num: Int) -> Bool {
    if num < 2 {
        return false
    }
    
    let root = Int(pow(Double(num), 0.5))
    for i in 2..<root+1 {
        if num % i == 0 {
            return false
        }
    }
    return true
}

func solution(_ nums:[Int]) -> Int {
    var answer = 0

    for i in nums.indices {
        for j in i+1..<nums.count {
            for k in j+1..<nums.count {
                if isPrime(num: nums[i] + nums[j] + nums[k]) {
                    answer += 1
                }
            }
        }
    }
    
    return answer
}
